### PR TITLE
Feature: Saga StateMachine Engine and Designer support Groovy Script Task

### PR DESCRIPTION
**Pull request by long187**
### Ⅰ. Describe what this PR did
Feature: Saga StateMachine Engine and Designer support Groovy Script Task.

ScriptTask is used to handle complex parameters,
StateMachine Engine will not record state logs while ScriptTask executing, and its result will  not be used for transaction consistency judgment.

### Ⅳ. Describe how to verify it
json definition:
```
       "ScriptState": {
            "Type": "ScriptTask",
            "ScriptType": "groovy",
            "ScriptContent": "if(throwException){ throw new RuntimeException(\"test\") } else { 'hello ' + inputA }",
            "Input": [
                {
                    "inputA": "$.[a]",
                    "throwException": "$.[scriptThrowException]"
                }
            ],
            "Output": {
                "scriptStateResult": "$.#root"
            },
            "Catch": [
                {
                    "Exceptions": [
                        "java.lang.Throwable"
                    ],
                    "Next": "Fail"
                }
            ],
            "Next": "ChoiceState"
        }
```

designer screenshot:
屏幕快照 2020-06-19 下午2 08 50

Unit test see:
StateMachineTests#testSimpleScriptTaskStateMachine
StateMachineDBMockServerTests#testSimpleScriptTaskStateMachineWithLayout

seata/test/src/test/resources/saga/statelang/simple_statelang_with_script.json
seata/test/src/test/resources/saga/statelang/designer_simple_statelang_with_script.json

### CODE REVIEW

**Comment by wangliang1986** [comment]
赞一个，哈哈。

File: saga/seata-saga-engine/src/main/java/io/seata/saga/engine/pcext/handlers/ScriptTaskStateHandler.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                        }
+                        else {
+                            //if we do not bind the null value property, groovy will throw MissingPropertyException
+                            bindings.put(property, null);
+                        }
+                    }
+                }
+            }
+            if (bindings != null) {
+                result = scriptEngine.eval(scriptContent, bindings);
```
Comment (wangliang1986) [inline saga/seata-saga-engine/src/main/java/io/seata/saga/engine/pcext/handlers/ScriptTaskStateHandler.java:105]: Will this script be parsed every time？

Comment (long187) [reply saga/seata-saga-engine/src/main/java/io/seata/saga/engine/pcext/handlers/ScriptTaskStateHandler.java:105]: I have cached ScripteEngine for each script type, and ScripteEngine will cache compiled scripts by default.

Comment (wangliang1986) [reply saga/seata-saga-engine/src/main/java/io/seata/saga/engine/pcext/handlers/ScriptTaskStateHandler.java:105]: OK.

**Pull request merged**